x-pack/plugins/reporting/server/core.ts

```diff
@@ -144,6 +149,20 @@ export class ReportingCore {
     this.pngExport = new PngExportType(this.core, this.config, this.logger, this.context);
     this.exportTypesRegistry.register(this.pngExport);
 
+    // deprecated export types for tests
+    this.pdfV1Export = new PdfV1ExportType(this.core, this.config, this.logger, this.context);
+    this.pngV1Export = new PngV1ExportType(this.core, this.config, this.logger, this.context);
+    this.csvSearchSourceImmediateExport = new CsvSearchSourceImmediateExportType(
+      this.core,
+      this.config,
+      this.logger,
+      this.context
+    );
+    this.exportTypesRegistry.register(this.pdfV1Export);
+    this.exportTypesRegistry.register(this.pngV1Export);
+    // @ts-ignore known issue with csv searchsource immediate
+    this.exportTypesRegistry.register(this.csvSearchSourceImmediateExport);
```

Comment:
I think it will be acceptable to make the `this.csvSearchSourceImmediateExport` only available to the area that needs it, which is `server/routes/generate/csv_searchsource_immediate.ts`;

This object was registered into the registry in the previous code, but it has never been a good fit, and has caused complications in the code.

Instead of collecting this object from the registry in `server/routes/generate/csv_searchsource_immediate.ts`, I propose that we add a method to reportingCore to access it there.